### DISCUSSION TITLE
Setup with 3.3.25

### DISCUSSION THREAD

**Discussion by sroycoursera** [Tyler-Keith-Thompson/CucumberSwift#65] [Q&A] [answered]
Hey everyone, I'm trying to set up CucumberSwift version 3.3.25 with Xcode 14.2 and not having success so far. Are other people experiencing this?

I follow the steps [from the SPM tutorial](https://tyler-keith-thompson.github.io/CucumberSwift/tutorials/cucumberswift/spm-step-by-step). I actually tried a couple times in case I missed something. When I run the test suite, it passes but doesn't detect or run any test. By contrast, [the sample project](https://github.com/Tyler-Keith-Thompson/CucumberSwiftSample/tree/main/SwiftPackageManager) works fine and I see the CucumberTest case detected and listed in the Test navigator of Xcode. The difference I see is that the sample uses CucumberSwift 3.1.1.

Anyone has some tips for getting it to behave? Should I just downgrade?

**Comment by Tyler-Keith-Thompson** [comment, marked as answer]
Is it not even running "testGherkin"? Usually this is just because your features folder is misnamed (it's case sensitive!). There were some diagnostics added recently that were supposed to help.

**Reply by sroycoursera** [reply]
I tried it again from scratch today. It doesn't appear to be picking up testGherkin at all.

- I verified that the Features folder is correctly named. I even tried by explicitly adding FeaturesPath in the Info tab, even though I'm using the default name. No go.
- I don't see a GenerateStepStubsIfNecessary step in the build or log.
- I made sure that the `.feature` file belongs to the UITests target and that it's listed in the Copy Bundle Resources build phase.
- I remembered seeing some odd behaviour related to folder references so I tried importing as a group instead. Still no go.

I'm not sure what else to look at so here are some screen shots. Any ideas? 🤔

<img width="1512" alt="Screenshot 2023-02-01 at 4 51 59 PM" src="https://user-images.githubusercontent.com/99687492/216174165-e114e3ac-745f-4b17-9225-dfaadad62061.png">
<img width="1512" alt="Screenshot 2023-02-01 at 4 51 19 PM" src="https://user-images.githubusercontent.com/99687492/216174192-db23adab-2fb6-42a6-ab1f-b4b25905c45c.png">
<img width="1512" alt="Screenshot 2023-02-01 at 4 51 34 PM" src="https://user-images.githubusercontent.com/99687492/216174235-964bb7bd-e5d8-4c0d-8f2d-d254768e9c4a.png">
<img width="1512" alt="Screenshot 2023-02-01 at 4 56 17 PM" src="https://user-images.githubusercontent.com/99687492/216174282-29f0dfbc-b438-42a3-90b8-053b709c667a.png">

**Reply by Tyler-Keith-Thompson** [reply]
Lame! Alright, as soon as I get a moment I will try to repro this by starting a new project from scratch. I can see you've got a library linked, but it's behaving like CucumberSwift isn't linked...See what's odd here is that the `testGherkin` that should show up is literally just an XCTest like you'd write yourself and hasn't changed in quite a while. 

You can try downgrading, but I'd be surprised if that solved anything. Still, it might be a data point. 

I'm swamped right now, but given you're stuck at trying to get *anything* to run, I will make this a high priority item and try to get you unblocked, thanks for reporting this. 

What version of macOS are you running? Just in case Apple did something silly with Xcode that's tied to the OS (they do that sometimes)

**Reply by sroycoursera** [reply]
I tried with 3.1.1 and also with a UIKit project instead of SwiftUI. Same pickle no matter what.

I'm using Xcode 14.2 on macOS Ventura 13.1.

Thanks for your help!

**Reply by Tyler-Keith-Thompson** [reply]
Just so you know I haven't forgotten about you. I appreciate your patience. I'm carving out time to repro your problem this weekend, or if I can't, send you a video of exactly what I do.

**Reply by Tyler-Keith-Thompson** [reply]
😱 I was able to repro, Apple must've broken the test discovery I was relying on. Working on a fix.

**Reply by Tyler-Keith-Thompson** [reply]
This is fixed in 4.0.1, Please adapt your code to have some class that inherits from CucumberTest, try this:
```swift
public var bundle: Bundle {
    class TestDiscovery: CucumberTest { }
    return Bundle(for: TestDiscovery.self)
}
```

**Reply by sroycoursera** [reply]
Great, I'll give it a shot later today and let you know!

**Reply by sroycoursera** [reply]
That got it over that hump and I was able to copy the generated steps. Thank you!

The Test navigator shows AFeatureWeWouldLikeToDescribe three times, once for each step, which is different than the sample project (still using 3.1.1) that only shows it once with the three steps within it. Is that expected?

<img width="373" alt="Screenshot 2023-02-06 at 1 12 50 PM" src="https://user-images.githubusercontent.com/99687492/217056198-a5f536af-6805-4241-92b1-57317ce47bb5.png">

**Reply by Tyler-Keith-Thompson** [reply]
That might be related to https://github.com/Tyler-Keith-Thompson/CucumberSwift/pull/73 which was a quite recent change. The idea behind that PR was to group by scenario. It's possible the test navigator and the report navigator look different, it's also possible a bug was introduced.
